client/app/dbaas/logs/detail/streams/alerts/streams-alerts.service.js

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @param {any} alertId - the alert ID for which alert object is to be fetched
+     * @returns promise which will be resolve with the alert
+     * @memberof StreamsAlertsService
+     */
+    getAlert (serviceName, streamId, alertId) {
+        return this.AlertsApiService.get({ serviceName, streamId, alertId }).$promise;
+    }
+
+    /**
+     * Returns a new alert object with the default properties
+     *
+     * @param {any} conditionType - the type of the condition (one of StreamsAlertsConstant.alertType)
+     * @returns the default alert object
+     * @memberof StreamsAlertsService
+     */
+    getNewAlert (conditionType) {
+        const thresholdType = conditionType === this.StreamsAlertsConstant.alertType.numeric ?
+            this.StreamsAlertsConstant.thresholdType.lower :
+            this.StreamsAlertsConstant.thresholdType.more;
+        const constraintType = this.StreamsAlertsConstant.constraintType.mean;
+        return {
+            conditionType,
+            thresholdType,
+            threshold: 1,
+            time: 1,
+            grace: 1,
+            backlog: 1,
+            repeatNotificationsEnabled: false,
+            constraintType
+        };
+    }
+
+    /**
+     * handles success state for create and delete alerts.
+     * Repetedly polls for operation untill it returns SUCCESS message.
+     *
+     * @param {any} serviceName
+     * @param {any} operation, operation to poll
+     * @param {any} successMessage, message to show on UI
+     * @returns promise which will be resolved to operation object
+     * @memberof LogsStreamsService
+     */
+    _handleSuccess (serviceName, operation, successMessage) {
+        this.poller = this._pollOperation(serviceName, operation);
+        return this.poller.$promise
+            .then(this.ServiceHelper.successHandler(successMessage));
+    }
+
+    /**
+     * Kills the current poller
+     *
+     * @memberof LogsStreamsService
+     */
+    _killPoller () {
+        if (this.poller) {
+            this.poller.kill();
+        }
+    }
+
+    /**
+     * Sets up polling for an operation
+     *
+     * @param {any} serviceName
+     * @param {any} operation, operation to poll
+     * @returns returns the poller promise that resolves when the polling for success is complete
+     * @memberof LogsStreamsService
+     */
+    _pollOperation (serviceName, operation) {
+        this._killPoller();
+        const poller = this.CloudPoll.poll({
+            item: operation,
+            pollFunction: opn => this.OperationApiService.get({ serviceName, operationId: opn.operationId }).$promise,
+            stopCondition: opn => opn.state === "FAILURE" || opn.state === "SUCCESS"
+        });
+        return poller;
+    }
+
+    /**
+     * Cleans the cache
+     *
+     * @memberof LogsStreamsService
+     */
+    _resetAllCache () {
+        this.AlertsApiService.resetAllCache();
```

Comment:
As stated before, this shouldn't be needed.

Reply:
Redundant cache clear is removed